### PR TITLE
 Planning: Costheta reference line smoother improvements

### PR DESCRIPTION
**Pull request by JasonZhou404**
Not enabled.

### CODE REVIEW

File: modules/planning/reference_line/cos_theta_problem_interface.cc [hunk 1]
```diff
@@ -555,5 +619,196 @@ void CosThetaProbleminterface::hessian_strcuture() {
...
+    g[index] = x[index];
+    g[index + 1] = x[index + 1];
+  }
+  return true;
+}
+
+/** Method to generate the required tapes */
+void CosThetaProbleminterface::generate_tapes(int n, int m, int* nnz_jac_g,
+                                              int* nnz_h_lag) {
+  Ipopt::Number* xp = new double[n];
```
Comment (xiaoxq) [inline modules/planning/reference_line/cos_theta_problem_interface.cc]: Avoid old-school memory management.
Use `vector<double> xp;`, and get the raw pointer as `&xp[0]` if the upstream API requires.

Comment (JasonZhou404) [reply modules/planning/reference_line/cos_theta_problem_interface.cc]: Part of them fixed. Some of them are not fixed successfully yet, will look into it afterwards

File: modules/planning/reference_line/cos_theta_reference_line_smoother.h [hunk 2]
```diff
@@ -43,25 +44,31 @@ class CosThetaReferenceLineSmoother : public ReferenceLineSmoother {
   void SetAnchorPoints(const std::vector<AnchorPoint>&) override;
 
  private:
-  bool Smooth(const std::vector<Eigen::Vector2d>& point2d,
-              const std::vector<double>& lateral_bounds,
-              std::vector<common::PathPoint>* ptr_smoothed_point2d);
+  bool CosThetaSmooth(const std::vector<Eigen::Vector2d>& point2d,
```
Comment (YajiaZhang) [inline modules/planning/reference_line/cos_theta_reference_line_smoother.h]: Better rename it to smooth; the function is defined in CosThetaReferenceLineSmoother, no need to specifically state CosThetaSmooth.

**Comment by xiaoxq** [comment]
CI seems stuck. I'll force-merge it per committer request.

**Pull request merged**